### DISCUSSION TITLE
Is expo router ready for production?

### DISCUSSION THREAD

**Discussion by kevnoutsawo** [expo/router#603] [Q&A] [answered]
Hello,
I recently discovered expo router and I am trying it out. So far there are a lot of undocumented/unimplemented features. It is understandable as it is pretty new. I also see that trying to use expo router only for file-based routing and using react navigation for layout and other functions creates conflicts as expo router uses react navigation under the hood.
We are supposed to be able to resort to react navigation directly when the concerned functionality is not in yet available as said here: 

> In [React Navigation v7](https://twitter.com/reactnavigation/status/1626648568189620238?s=20) a number of concepts from 
> Expo Router like Link href and static configuration have been upstreamed, this helps keep improve the ability to drop down 
> from Expo Router when required functionality is not yet available.

But I couldn't find in the documentation how that should be done, especially how to avoid conflicts with expo router's underlying react navigation uses.

> # RFC
> The entire contents of this repo are up for discussion. We are releasing the beta early and in a very volatile state! Please do not depend on this library in your production apps yet!

Is the official position still against using expo router in production?

Thanks @EvanBacon and all the team for this library, this paradigm shift is very welcome on mobile.

**Comment by EvanBacon** [comment, marked as answer]
- The API is stable.
- There are known bugs that we're working on.
- File-based routing on native is new and therefore means you won't find as much documentation around it as you might find for something more established.
- The kick app, which chart-topped in the Apple App Store was using Expo Router last time I checked.
